### DISCUSSION TITLE
CobraPin Getting Started

### DISCUSSION THREAD

**Discussion by ttrickel** [missionpinball/missionpinball#68] [Q&A] [answered]
I have a Gen2 CobraPin board and I'm trying to get it working with MPF.

I can run MPF and "see" switches closing when I press the assigned keyboard keys.

I have a different config.yaml file but when that failed to light yellow LEDs for the coils I thought I'd strip it back to a more basic level.

I found the following config.yaml on the [MPF CobraPin page](https://docs.missionpinball.org/en/latest/hardware/opp/cobrapin/index.html).

Should I expect the 1-0-1 yellow LED coil light to flash if I use the following as a config.yaml file?

#config_version=5

hardware:
   platform: opp
   driverboards: gen2

opp:
   ports: com4, com5
   baud: 115200
   debug: True

coils:
    c_my_coil:
       number: 1-0-1
       pulse_events: s_my_switch_active

switches:
   s_my_switch:
      number: 0-0-16

**Comment by ralfit0** [comment, marked as answer]
ttrickel,
if that is your complete config.yaml, you are missing (at least) a device section.
Suggest to follow the MPF tutorial step by step, especially https://docs.missionpinball.org/en/latest/tutorial/3_get_flipping.html
Also: "Note that if you have a physical machine connected, your physical flippers will not flip with the keyboard keys."
(so you need to trigger them with a real hardware switch at 0-0-16)

**Reply by ttrickel** [reply]
Thank you for your response ralfit0,

Your note is what I really needed to read. Does this extend to the rest of the "machine" or is it limited to coil devices?

I have been running through the tutorial. I'm up to step 12 and thought I'd try the real machine.

Who do I email about changes to a web page? 

I think the page: [CobraPin Pinball Controller powered by OPP](https://docs.missionpinball.org/en/latest/hardware/opp/cobrapin/index.html) should be changed where it says: "To run the above test ..." to include the switch is necessary and a working config file. I'd be more than willing to make the changes myself. 

Is there a process for this?

Sorry about all the questions.

**Comment by ttrickel** [comment]
I confirmed that with the addition of a physical switch connected to 0-0-16, as ralfit0 said, the config.yaml in my question will turn on one of the yellow coli LEDs on a CobraPin board when the switch is pressed. No devices section needed for this very limited test.

**Comment by ralfit0** [comment]
> Is there a process for this?

You bet!  https://docs.missionpinball.org/en/stable/about/contributing_to_mpf_docs.html
